extensions/interactions/AlgebraicExpressionInput/directives/algebraic-expression-input-rules.service.spec.ts

```diff
@@ -29,218 +29,239 @@ describe('Algebraic expression input rules service', () => {
   });
 
   it('should have a correct MatchesExactlyWith rule', () => {
-    inputString = '((x)^(2)-x)/(z)-4*y';
+    inputString = '(x^2-x)/z-4*y';
```

Comment:
Not sure this needs to be changed btw.

Also doesn't guppy automatically stick parens around things in its internal representation? If so, then forbidding the paren version might be an issue.

Reply:
Ah, yes, that's right! We can't restrict single variable enclosing parens then. I don't think there's any case where guppy will auto add multiple consecutive params so they can be restricted.